sam.c

```diff
@@ -3960,7 +4018,7 @@ void bam_plp_destructor(bam_plp_t plp,
  *  Returns BAM_CMATCH, -1 when there is no more cigar to process or the requested position is not covered,
  *  or -2 on error.
  */
-static inline int cigar_iref2iseq_set(uint32_t **cigar, uint32_t *cigar_max, int *icig, int *iseq, int *iref)
+static inline int cigar_iref2iseq_set(uint32_t **cigar, uint32_t *cigar_max, int *icig, int *iseq, hts_pos_t *iref)
 {
     int pos = *iref;
```

Comment:
64 bit to 32 bit conversion.  Is this right?

Reply:
I think it gets away with it because `*iref` is set to an offset to the beginning of the read, so should fit within an `int` (barring issues with very big `BAM_CDEL` or `BAM_CREF_SKIP` CIGAR operations).  However it might be better to make `cigar_iref2iseq_set`, `cigar_iref2iseq_next` and `tweak_overlap_quality`, which calls them, use `hts_pos_t` all round as avoiding any possibility of overflow will make the code easier to understand.